test/dotnet-new3.UnitTests/DotnetNewSearch.cs

```diff
@@ -463,5 +495,39 @@ private List<List<string>> ParseTableOutput(string stdOut, string[] expectedColu
             }
             return parsedTable;
         }
+
+        private class ShrinkAwareOrdinalStringComparer : IComparer<string>
+        {
+            public int Compare(string? left, string? right)
+            {
+                if (string.IsNullOrEmpty(left) && string.IsNullOrEmpty(right))
+                {
+                    return 0;
+                }
+
+                if (string.IsNullOrEmpty(left))
+                {
+                    return -1;
+                }
+
+                if (string.IsNullOrEmpty(right))
+                {
+                    return 1;
+                }
+
+                bool leftIsShrunk = left.EndsWith("...");
+                bool rightIsShrunk = right.EndsWith("...");
+                if (leftIsShrunk || !rightIsShrunk)
+                {
+                    return string.Compare(left, right, StringComparison.Ordinal);
+                }
+
+                if (left.StartsWith(right.Substring(0, right.Length - 3), StringComparison.Ordinal))
+                {
+                    return -1;
+                }
+                return string.Compare(left, right, StringComparison.Ordinal);
```

Comment:
According to this comparer:
- "b..." is less than "ba" 
and
- "ba" is less than "b..."

at the same time. 

Could this be updated so that for any two strings `x` and `y`, if `x < y`, then `y > x` holds true? Or is this somehow intended?

Reply:
Not sure if we can do any better here... We don't know what "b..." is: it can be "booo" and it can be "baaa".  So if it's shrunk and other part contains it we need to return true as we don't know.

I've made the comparer more generic, so it works for both sides now.